external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaTupleListener.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.storm.kafka.spout;
+
+import java.io.Serializable;
+import java.util.List;
+
+
+/**
+ * The KafkaTupleListener handles state changes of a kafka tuple inside a KafkaSpout.
+ */
+public interface KafkaTupleListener extends Serializable {
```

Comment:
Good idea, I have added a prepare method or "open" since it is in the context of a spout.  

I am not sure what you mean by "dead tuples". 
If `isEmitTuple(tuple)` is false in line 380 of the KafkaSpouts `emitTupleIfNotEmitted`? But then i would have to change the onAck call as well, since it would be called twice. I can imagine that someone needs more callbacks. However, I tried to keep the interface simple and small. But if you wish I would gladly add new functionality to the interface.